src/number/complete.rs

```diff
@@ -1603,6 +1690,78 @@ where
   Ok((i, float))
 }
 
+/// Recognizes floating point number in text format and returns a f64.
+/// This only handles non-finite (special) values.
+pub fn double_nonfinite<T, E: ParseError<T>>(input: T) -> IResult<T, f64, E>
+where
+  T: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>> + Slice<Range<usize>>,
+  T: Clone + Offset,
+  T: InputIter + InputLength + InputTake,
+  <T as InputIter>::Item: AsChar + Copy,
+  <T as InputIter>::IterElem: Clone,
+  T: InputTakeAtPosition,
+  <T as InputTakeAtPosition>::Item: AsChar,
+  T: AsBytes,
+  T: for<'a> Compare<&'a [u8]>,
+{
+  let (i, sign) = sign(input.clone())?;
+  let (mut double, count) = if i.input_len() >= 3 {
+    if case_insensitive_cmp(i.as_bytes(), b"nan") {
+      (f64::NAN, 3)
+    } else if i.input_len() >= 8 && case_insensitive_cmp(i.as_bytes(), b"infinity") {
+      (f64::INFINITY, 8)
+    } else if case_insensitive_cmp(i.as_bytes(), b"inf") {
+      (f64::INFINITY, 3)
+    } else {
+      return Err(Err::Error(E::from_error_kind(input, ErrorKind::Float)));
+    }
+  } else {
+    return Err(Err::Error(E::from_error_kind(input, ErrorKind::Float)));
+  };
+
+  if !sign {
+    double = -double;
+  }
+
+  Ok((i.slice(count..), double))
+}
+
+/// Recognizes floating point number in text format and returns a f64.
+///
+/// *Complete version*: Can parse until the end of input.
+/// ```rust
+/// # use nom::{Err, error::ErrorKind, Needed};
+/// # use nom::Needed::Size;
+/// use nom::number::complete::double;
+///
+/// let parser = |s| {
+///   double(s)
+/// };
+///
+/// assert_eq!(parser("11e-1"), Ok(("", 1.1)));
+/// assert_eq!(parser("123E-02"), Ok(("", 1.23)));
+/// assert_eq!(parser("123K-01"), Ok(("K-01", 123.0)));
+/// assert_eq!(parser("abc"), Err(Err::Error(("abc", ErrorKind::Float))));
+/// ```
+pub fn double<T, E: ParseError<T>>(input: T) -> IResult<T, f64, E>
+where
+  T: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>> + Slice<Range<usize>>,
+  T: Clone + Offset,
+  T: InputIter + InputLength + InputTake,
+  <T as InputIter>::Item: AsChar + Copy,
+  <T as InputIter>::IterElem: Clone,
+  T: InputTakeAtPosition,
+  <T as InputTakeAtPosition>::Item: AsChar,
+  T: AsBytes,
+  T: for<'a> Compare<&'a [u8]>,
+{
+  match double_finite::<T, E>(input.clone()) {
+    Ok((i, double)) => Ok((i, double)),
+    Err(Err::Incomplete(e)) => Err(Err::Incomplete(e)),
```

Comment:
impossible in complete variant

Reply:
Whoops my bad, thanks for the catch.

Reply:
Removed as of the latest commit, this should be resolved.